### PR TITLE
Split Cekit.run into run, configure

### PR DESCRIPTION
**Pull request by jmtd**
Split the run() method into a configure() method and a run()
method that calls it.

This means we can write tests for the configure() code without
invoking run() and generating Dockerfiles etc.

(Review tip: try ignoring whitespace changes with "git show HEAD -w")

### CODE REVIEW

File: cekit/cli.py [hunk 1]
```diff
@@ -171,31 +171,36 @@ def run(self):
...
+                                   self.args.target,
+                                   self.args.build_engine,
+                                   self.args.overrides,
+                                   params)
 
-            # We have all overrided repo fetch so we can discover modules
-            # and process its dependency trees
-            discover_modules(os.path.join(self.args.target, 'repo'))
+        # Now we can fetch repositories of modules (we have all overrides)
+        get_dependencies(self.generator.image,
```
Comment (dbecvarik) [inline cekit/cli.py]: this is  cloning the module repos I would postpone that to run method.

**Comment by jmtd** [comment]
The motivation for this PR is to make writing tests in #272 easier

File: cekit/cli.py (hunk 1 above)
Comment (jmtd) [reply cekit/cli.py]: Thanks, good point: force-pushed a rebase to fix.

**Comment by dbecvarik** [comment]
thanks!

**Pull request merged**